src/proto/stream.rs

```diff
@@ -340,7 +340,14 @@ where
             return Err(Error::AlreadyClosed);
         }
 
-        if item.opcode.is_control() || item.payload.len() <= self.config.frame_size {
+        let is_control = item.opcode.is_control();
+        if is_control || item.payload.len() <= self.config.frame_size {
+            if is_control && item.payload.len() > 125 {
```

Comment:
I would rather have this check when creating control messages @Gelbpunkt

Reply:
Yeah, that would generally make more sense from a design standpoint. We can assert the length in the respective methods

Reply:
Thanks for the review, I've made this change :)